src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -1693,16 +1692,14 @@ static PyObject* seg_point_processes(NPySegObj* self) {
     Section* sec = self->pysec_->sec_;
     CHECK_SEC_INVALID(sec);
     Node* nd = node_exact(sec, self->x_);
-    PyObject* result = PyList_New(0);
+    nb::list result{};
     for (Prop* p = nd->prop; p; p = p->next) {
         if (memb_func[p->_type].is_point) {
             auto* pp = p->dparam[1].get<Point_process*>();
-            auto item = nb::steal(nrnpy_ho2po(pp->ob));
-            int err = PyList_Append(result, item.ptr());
-            assert(err == 0);
+            result.append(nb::steal(nrnpy_ho2po(pp->ob)));
         }
     }
-    return result;
+    return result.release().ptr();
```

Comment:
This section looks like a 1-to-1 translation. Very nice & AFAICT `nrnpy_ho2po` return a "new reference".